ethstorage/miner/l1_mining_api.go

```diff
@@ -152,7 +108,7 @@ func (m *l1MiningAPI) SubmitMinedResult(ctx context.Context, contract common.Add
 		return common.Hash{}, fmt.Errorf("failed to estimate gas: %w", err)
 	}
 	m.lg.Info("Estimated gas done", "gas", estimatedGas)
-	cost := new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasPrice)
+	cost := new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), predictedGasPrice)
```

Comment:
That would result in negative profit if the gas price exceeds the predicted value. What’s your opinion on this situation? Can the users afford to ignore these cases?

Reply:
Yes, I believe we should do better to prevent miners from any loss. Added detailed updates in the description.